### PR TITLE
Fix a lot of clippy errors [part 1]

### PR DESCRIPTION
**Pull request by evgenykuzyakov**

### CODE REVIEW

File: chain/network/src/types.rs [hunk 1]
```diff
@@ -56,6 +56,7 @@ impl TryFrom<Vec<u8>> for PeerId {
     }
 }
 
+#[allow(clippy::derive_hash_xor_eq)]
```
Comment (bowenwang1996) [inline chain/network/src/types.rs]: This should not be allowed

File: chain/pool/src/lib.rs [hunk 2]
```diff
@@ -10,17 +10,14 @@ pub use crate::types::Error;
 pub mod types;
 
 /// Transaction pool: keeps track of transactions that were not yet accepted into the block chain.
+#[derive(Default)]
```
Comment (bowenwang1996) [inline chain/pool/src/lib.rs]: Why do we need this?

File: core/primitives/src/utils.rs [hunk 3]
```diff
@@ -188,15 +188,27 @@ impl<T: fmt::Display> From<Option<T>> for DisplayOption<T> {
     }
 }
 
-/// Macro to either return value if the result is Ok, or exit function logging error.
```
Comment (bowenwang1996) [inline core/primitives/src/utils.rs]: Why is this comment deleted?

File: runtime/runtime/src/lib.rs [hunk 4]
```diff
@@ -252,12 +252,12 @@ impl Runtime {
...
         }
 
-        if !check_rent(&signer_id, &mut signer, &self.config, apply_state.epoch_length) {
+        if !check_rent(&signer_id, &signer, &self.config, apply_state.epoch_length) {
             return Err(format!("Failed to execute, because the account {} wouldn't have enough to pay required rent", signer_id).into());
         }
 
         match access_key.permission {
-            AccessKeyPermission::FullAccess => Ok(()),
+            AccessKeyPermission::FullAccess => (),
```
Comment (bowenwang1996) [inline runtime/runtime/src/lib.rs]: This looks suboptimal. Why not use `if let` here?

File: scripts/run_clippy.sh [hunk 5]
```diff
@@ -1,2 +1,3 @@
 #!/bin/bash
-cargo clippy --all  -- -A clippy::type-complexity -A clippy::needless-pass-by-value -A clippy::while-let-loop -A clippy::too-many-arguments -A clippy::unit_arg -A clippy::if_same_then_else -A clippy::collapsible_if -A clippy::useless-let-if-seq -A clippy::map-entry -D warnings -A clippy::implicit-hasher -A clippy::ptr-arg -A renamed-and-removed-lints -A clippy::needless-range-loop
+cargo clippy --all  -- -A clippy::type-complexity -A clippy::needless-pass-by-value -A clippy::while-let-loop -A clippy::too-many-arguments -A clippy::unit_arg -A clippy::if_same_then_else -A clippy::collapsible_if -A clippy::useless-let-if-seq -A clippy::map-entry -D warnings -A clippy::implicit-hasher -A clippy::ptr-arg -A renamed-and-removed-lints -A clippy::needless-range-loop -A clippy::unused_io_amount -A clippy::len-without-is-empty
```
Comment (bowenwang1996) [inline scripts/run_clippy.sh]: Why do we add `len-without-is-empty`?

Comment (evgenykuzyakov) [reply scripts/run_clippy.sh]: A bunch of classes has the `len` method, but don't have `is_empty()`. Should we implement it?

Comment (bowenwang1996) [reply scripts/run_clippy.sh]: Sure

File: runtime/runtime/src/lib.rs (hunk 4 above)
Comment (evgenykuzyakov) [reply runtime/runtime/src/lib.rs]: good point. It happened after a bunch of refactoring.

File: core/primitives/src/utils.rs (hunk 3 above)
Comment (evgenykuzyakov) [reply core/primitives/src/utils.rs]: oops

File: chain/network/src/types.rs (hunk 1 above)
Comment (evgenykuzyakov) [reply chain/network/src/types.rs]: fixed.

File: chain/pool/src/lib.rs (hunk 2 above)
Comment (evgenykuzyakov) [reply chain/pool/src/lib.rs]: replaces `new` without args.

**Pull request merged**